unity/Assets/Scripts/ArtGallery/Util/NaiveLighthouseToLighthouseVisibility.cs

```diff
@@ -0,0 +1,279 @@
+        ///     <paramref name="vertexes" />
+        ///     can seen by one other vertex in
+        ///     <paramref name="vertexes" />
+        /// </returns>
+        public static bool VisibleToOtherVertex(
+            List<Vector2> vertexes,
+            Polygon2D polygon)
+        {
+            // For each of the vertexes check if they are visible to at least
+            // one other vertex
+            foreach (Vector2 vertex1 in vertexes)
+            {
+                var otherVertexes = vertexes.Where(i => i != vertex1).ToList();
+
+                // If the vertex cannot be seen by one other vertex return 
+                if (!VisibleToOtherVertex(vertex1, otherVertexes, polygon))
+                {
+                    return false;
+                }
+            }
+
+            // if every vertex can be seen by one other vertex return true
+            return true;
+        }
+
+        /// <summary>
+        ///     Creates a dictionary containing an entry for each vertex in
+        ///     <paramref name="vertexes" />
+        ///     and a corresponding value with all the vertexes in
+        ///     <paramref name="vertexes" />
+        ///     each vertex can see
+        /// </summary>
+        /// <param name="vertexes">
+        ///     The collection of vertexes for which visibility needs to be
+        ///     calculated
+        /// </param>
+        /// <param name="polygon">The polygon in which the vertexes exist.</param>
+        /// <returns>
+        ///     A dictionary containing for each vertex the other visible
+        ///     vertexes
+        /// </returns>
+        public static IDictionary<Vector2, ICollection<Vector2>>
+            VisibleToOtherVertexes(
+                List<Vector2> vertexes,
+                Polygon2D polygon)
+        {
+            // Create dictionary to store the result
+            IDictionary<Vector2, ICollection<Vector2>> result =
+                new Dictionary<Vector2, ICollection<Vector2>>();
+
+            // Iterate over all the vertexes and calculate for each vertex the
+            // other vertexes it can see.
+            foreach (Vector2 vertex1 in vertexes)
+            {
+                // Select all vertexes except the current vertex 
+                var otherVertexes = vertexes.Where(i => i != vertex1).ToList();
+
+                // Calculate the visible vertexes 
+                var visibleVertices = VisibleToOtherVertexes(
+                    vertex1,
+                    otherVertexes,
+                    polygon);
+
+                // Create a dictionary item. The key is the current vertex and
+                // the value is the vertexes it can see.
+                var dicItem =
+                    new KeyValuePair<Vector2, ICollection<Vector2>>(
+                        vertex1,
+                        visibleVertices);
+
+                // Add the dictionary item to the dictionary
+                result.Add(dicItem);
```

Comment:
Can't we just add use `result.Add(vertex, visibleVertices)` here?

Reply:
Yes we could. Changed it in code